### PR TITLE
fix(transformers): make sure to set correct GPU device on set framework

### PR DESCRIPTION
**Pull request by aarnphm**
cc @bojiang

Signed-off-by: Aaron <29749331+aarnphm@users.noreply.github.com>

### CODE REVIEW

File: src/bentoml/_internal/frameworks/transformers.py [hunk 1]
```diff
@@ -886,19 +900,56 @@ def __init__(self):
...
+                        with tf.device(
+                            "/device:CPU:0"
+                            if available_gpus in ("", "-1")
+                            else f"/device:GPU:{available_gpus}"
+                        ):
+                            self.model = t.cast(
+                                transformers.TFPreTrainedModel,
+                                load_model(bento_model, **kwargs),
+                            )
+                    else:
```
Comment (bojiang) [inline src/bentoml/_internal/frameworks/transformers.py:939]: shall we check if it is "JAX" here?

File: src/bentoml/_internal/frameworks/transformers.py [hunk 2]
```diff
@@ -886,19 +900,56 @@ def __init__(self):
...
+                            transformers.PreTrainedModel,
+                            load_model(bento_model, **kwargs),
+                        ).to(
+                            torch.device(
+                                "cuda" if available_gpus not in ("", "-1") else "cpu"
+                            )
+                        )
+                        torch.set_default_tensor_type("torch.cuda.FloatTensor")
+                    elif "tf" == bento_model.info.metadata["_framework"]:
+                        with tf.device(
```
Comment (bojiang) [inline src/bentoml/_internal/frameworks/transformers.py:930]: shall we delay the current tf importing to here?

File: src/bentoml/_internal/frameworks/transformers.py (hunk 1 above)
Comment (aarnphm) [reply src/bentoml/_internal/frameworks/transformers.py:939]: it is

File: src/bentoml/_internal/frameworks/transformers.py (hunk 2 above)
Comment (aarnphm) [reply src/bentoml/_internal/frameworks/transformers.py:930]: tf and torch is already lazy loaded

File: src/bentoml/_internal/frameworks/transformers.py (hunk 1 above)
Comment (aarnphm) [reply src/bentoml/_internal/frameworks/transformers.py:939]: since the possible runtime are just torch, tf and flax

**Review by bojiang — APPROVED** [review]

**Pull request merged**